lib/awful/tag.lua

```diff
@@ -746,9 +763,47 @@ function tag.object.set_layout(t, layout)
 
     tag.setproperty(t, "layout", layout)
 
+    local override = tag.getproperty(t, "layouts")
+
+    if not override then
+        t:emit_signal("property::layouts")
+    end
+
     return layout
 end
 
+function tag.object.get_layouts(self)
+    local override = tag.getproperty(self, "layouts")
+
+    if override then
+        return override
+    end
+
+    -- The override has the priority.
+    local cls = tag.getproperty(self, "custom_layouts")
+
+    -- Allow custom layouts to be listed until GC-ed.
+    if not cls then
+        cls = setmetatable({}, {__mode="v"})
+        tag.setproperty(self, "custom_layouts", cls)
+    end
+
+    -- Required to get the default/fallback list of layouts
+    alayout = alayout or require("awful.layout")
+
+    -- Add the custom and unlisted layouts to te front
+    if (not gtable.hasitem(cls, self.layout))
+      and not alayout.get_tag_layout_index(self) then
+        table.insert(cls, self.layout)
```

Comment:
Which piece of code is responsible for removing this item again?
(And the answer "the garbage collector" still means: Why is it okay for the list of layouts to change randomly? (E.g.: No signal is emitted))